cmds/webboot/webboot.go

```diff
@@ -66,11 +78,50 @@ func (d *DownloadByLink) Exec() error {
 	return nil
 }
 
+// Exec displays subdirectory or cached isos under the path directory
+func (g *DirGroup) Exec() error {
+
+	entries := []menu.Entry{}
+
+	readerInfos, err := ioutil.ReadDir(g.path)
+	if err != nil {
+		return err
+	}
+
+	// check the directory, if there is a subdirectory, add another DirGroup option
+	// if there is iso file, add a chacedIso option
+	for _, info := range readerInfos {
+		if info.IsDir() {
+			entries = append(entries, &DirGroup{
+				name:     info.Name(),
+				path:     filepath.Join(g.path, info.Name()),
+				uiEvents: g.uiEvents,
+			})
+		} else if filepath.Ext(info.Name()) == ".iso" {
+			iso := &CachedISO{
+				path:  filepath.Join(g.path, info.Name()),
+				label: info.Name(),
+			}
+			entries = append(entries, iso)
+		}
+	}
+	_, err = menu.DisplayMenu("Groups", "Choose a group of chached iso", entries, g.uiEvents)
```

Comment:
I think the menu could just be titled something like "Distros" and just say something like "Choose an option:" - I don't think the group terminology is clear